math.go

```diff
@@ -14,12 +17,12 @@ func dct(width int, height int, values []uint8) (result []float32) {
 				for j := 0; j < width; j++ {
 
 					sum += float64(values[i*width+j]) *
-						math.Cos(((math.Pi*float64(u))/(2*float64(height)))*(2*float64(i)+1)) *
-						math.Cos(((math.Pi*float64(v))/(2*float64(width)))*(2*float64(j)+1))
+						math.Cos(((math.Pi*float64(u))/doubleHeight)*(2*float64(i)+1)) *
+						math.Cos(((math.Pi*float64(v))/doubleWidth)*(2*float64(j)+1))
 				}
 			}
 
-			result[u*width+v] = float32(sum)
+			result[u*width+v] = int16(sum)
```

Comment:
Got the color representation relevance :+1: and that the results are indeed `int`s when using the example code (thanks for the example code!). I'm playing with it to understand how the coefficients in the DCT modify the basic functions (the images in the matrix).

----

I'm not familiar with gamma correction, but I saw an example and I think I got the main idea of what the correction is doing, but maybe not what we would use it for.

Would that act as a sort of normalization, with a potentially different factor for each image? Would we use it as a way to make sure that we take into account as many details in the image as the "_luminance? range_" allows us to encode? (I'm not sure if the  _luminance_ is what we're extracting when converting the original image to a gray scale.)

I observe that two plain RGB `#888820` and RGB `#638888` images have the same DCT - the zero matrix using the example code - despite being at different "distances" from `#888888` in terms of how much color was changed to go from `#888888` to each of them, resp. 88 to 20 and 88 to 63. (Does that make sense?) And I imagine that different colors are not contributing in the same way to the image _luminance_. (I'll just keep saying luminance, it may not be the right term.)

That makes me think that depending on the color range of the image, part of the gray scale (I think of it as the encoding space we have for the luminance) could happen not to be used. If I understand well, a carefully chosen gamma correction would expand the pixel values to occupy all that space. Am I close?

It will probably be easier to talk anyway, but I wanted to write the idea down before I forget something :P

(I'm picturing that to myself as using close-IR illumination for security cameras at night: the details are there, _we_ just don't see them. And it happens that the camera has some unused sensibility / "encoding space" in the close IR that can be leveraged of only we expand the image color range a little bit... (?) I might be getting both things wrong ^^)